packages/nimble-components/src/utilities/tests/component.ts

```diff
@@ -4,3 +4,28 @@ export async function clickElement(element: HTMLElement): Promise<void> {
     element.click();
     await waitForUpdatesAsync();
 }
+
+/** A helper function to abstract adding an event listener, spying
+ * on the event being called, and removing the event listener. The returned promise
+ * should be resolved prior to completing a test.
+ */
+export function createEventListener(
+    element: HTMLElement,
+    eventName: string
+): {
+        promise: Promise<void>,
+        spy: jasmine.Spy
+    } {
+    const spy = jasmine.createSpy();
+    return {
+        promise: new Promise(resolve => {
+            const handler = (...args: unknown[]): void => {
+                element.removeEventListener(eventName, handler);
+                spy(...args);
+                resolve();
+            };
+            element.addEventListener(eventName, handler);
+        }),
+        spy
+    };
+}
```

Comment:
nit: Feel free to ignore if want to merge soon, realized we can simplify it a tad:
```ts
export function createEventListener(
    element: HTMLElement,
    eventName: string
): Promise<jasmine.Spy> {
    const spy = jasmine.createSpy();
    return new Promise(resolve => {
            const handler = (...args: unknown[]): void => {
                element.removeEventListener(eventName, handler);
                spy(...args);
                resolve(spy);
            };
            element.addEventListener(eventName, handler);
        });
}
```

and then usage
```
                const toggleListener = createEventListener(
                    menuButton,
                    'toggle'
                );
await toggleListener; // don't care about the spy
const spy = await toggleListener;
```

Reply:
I do like that pattern, but it breaks some of the menu-button tests that need access to the spy before the promise resolves in order to assert that the `beforetoggle` event was emitted before the `toggle` event was emitted. I'd rather not try to come up with a different pattern for those tests in this PR, so I think I'll leave it as-is for now.

Reply:
Ah didn't think about that, makes sense to keep it as-is for now 👍